openedx/features/courseware/tests/test_pre_post_assessment_scores_api.py

```diff
@@ -0,0 +1,148 @@
+
+    def create_course_and_user(self):
+        """
+        Create user and a simple course for testing pre/post assessments score api.
+        """
+        self.course = CourseFactory.create(display_name='test_course', number='100')
+        self.user = UserFactory.create(username='dummy', password='foo')
+        self.add_pre_post_assessment_grading_policy()
+
+    def create_chapter(self):
+        """Create chapter in course"""
+        return ItemFactory.create(
+            parent_location=self.course.location,
+            category='chapter',
+            display_name='Chapter'
+        )
+
+    def create_assessment_subsection(self, chapter, assessment_format=constants.PRE_ASSESSMENT_FORMAT):
+        """Create pre/post assessment in module/chapter"""
+        return ItemFactory.create(
+            parent_location=chapter.location,
+            display_name=assessment_format,
+            category='sequential',
+            metadata={'graded': True, 'format': assessment_format}
+        )
+
+    def get_pre_post_assessment_api_response(self, chapter_id):
+        """Method to create url and call api"""
+        url = reverse(
+            'courseware:pre_post_assessments_score',
+            kwargs={
+                'course_id': self.course.id,
+                'chapter_id': chapter_id,
+            }
+        )
+
+        return self.client.get(url)
+
+    def add_pre_post_assessment_grading_policy(self):
+        """Add grading policy to mark assessments pre or post for testing"""
+        grading_policy = {
+            "GRADER": [
+                {
+                    "type": constants.PRE_ASSESSMENT_FORMAT,
+                    "min_count": 1,
+                    "drop_count": 0,
+                    "short_label": constants.PRE_ASSESSMENT_FORMAT,
+                    "weight": 0.0
+                },
+                {
+                    "type": constants.POST_ASSESSMENT_FORMAT,
+                    "min_count": 1,
+                    "drop_count": 0,
+                    "short_label": constants.POST_ASSESSMENT_FORMAT,
+                    "weight": 0.25
+                }
+            ]
+        }
+
+        self.course.grading_policy = grading_policy
+        self.update_course(self.course, self.user.id)
+        self.course = self.store.get_course(self.course.id)
+
+
+class PrePostAssessmentScoreForUnEnrolledUserTest(PrePostAssessmentMixin, APITestCase):
+    """Tests for unenrolled and unauthorized users"""
+
+    def setUp(self):
+        super(PrePostAssessmentScoreForUnEnrolledUserTest, self).setUp()
+        self.create_course_and_user()
+        self.client.force_authenticate(self.user)
+
+    def test_un_enrolled_user_request(self):
+        """Test unenrolled user request"""
+        response = self.get_pre_post_assessment_api_response(self.create_chapter().location.block_id)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_unauthorized_request(self):
```

Comment:
We are using `view_auth_classes` deco. in our API which uses `JwtAuthentication, OAuth2AuthenticationAllowInactiveUser, SessionAuthenticationAllowInactiveUser` under the hood and their tests are written by their respective packages, So we don't need to test those packages again.